isovar/isovar_result.py

```diff
@@ -574,21 +578,21 @@ def protein_sequence_contains_mutation(self):
 
         Returns bool
         """
-        if not self.has_mutant_protein_sequence_from_rna:
+        if self.top_protein_sequence is None:
             return None
+        return self.top_protein_sequence.contains_mutation
 
-        # if the sequence is the same as the reference then the genomic
-        # variant must have been silent, or maybe was made silent by adjacent
-        # phased variants.
-        if self.protein_sequence_matches_reference:
-            return False
-
-        start_idx  = self.protein_sequence_mutation_start
-        stop_idx = self.protein_sequence_mutation_end
+    @cached_property
+    def protein_sequence_contains_deletion(self):
```

Comment:
why is this special-cased?

Reply:
Since deletions don't really create new amino acid sequences but only count as mutations due to novel juxtaposition of amino acids to their left and right -- I want to be able to track that easily.